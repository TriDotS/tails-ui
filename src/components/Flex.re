let component = ReasonReact.statelessComponent("Flex");

let make =
    (~display, ~direction, ~wrap, ~justify, ~align, ~height, ~width, ~children) => {
  ...component,
  render: _self =>
    <div
      className={j| h-$(height) w-$(width) $(display) flex-$(direction) flex-$(wrap) justify-$(justify) items-$(align)|j}>
      (ReasonReact.arrayToElement(children))
    </div>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~display=jsProps##display,
      ~direction=jsProps##direction,
      ~wrap=jsProps##wrap,
      ~justify=jsProps##justify,
      ~align=jsProps##align,
      ~height=jsProps##height,
      ~width=jsProps##width,
      ~children=jsProps##children
    )
  );
