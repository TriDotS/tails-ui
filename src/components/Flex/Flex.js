import PropTypes from 'prop-types'
import React from 'react'

const Flex = ({
  children,
  display,
  direction,
  wrapping,
  justify,
  align,
  padding,
  margin
}) => (
  <div
    className={`${display} flex-${direction} flex-${wrapping} justify-${justify} items-${align} p-${padding} m-${margin}`}
  >
    {children}
  </div>
)

Flex.defaultProps = {
  display: 'flex',
  direction: 'col',
  wrapping: 'wrap',
  justify: 'start',
  align: 'start',
  padding: '0',
  margin: '0'
}

Flex.propTypes = {
  children: PropTypes.node.isRequired,
  display: PropTypes.oneOf(['flex', 'inline-flex']),
  direction: PropTypes.oneOf(['row', 'row-reverse', 'col', 'col-reverse']),
  wrapping: PropTypes.oneOf(['no-wrap', 'wrap', 'wrap-reverse']),
  justify: PropTypes.oneOf(['start', 'center', 'end', 'between', 'around']),
  align: PropTypes.oneOf(['stretch', 'start', 'center', 'end', 'baseline']),
  padding: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  margin: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8'])
}

export default Flex
