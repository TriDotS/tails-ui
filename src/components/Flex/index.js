import PropTypes from 'prop-types'
import React from 'react'

const Flex = ({
  display,
  direction,
  wrapping,
  justify,
  align,
  padding,
  margin,
  children,
  smHeight,
  mdHeight,
  lgHeight,
  xlHeight,
  smWidth,
  mdWidth,
  lgWidth,
  xlWidth
}) => (
  <div
    className={`h-${smHeight} sm:h-${smHeight} md:h-${mdHeight} lg:h-${lgHeight} xl:h-${xlHeight} w-${smWidth} sm:w-${smWidth} md:w-${mdWidth} lg:w-${lgWidth} xl:w-${xlWidth} ${display} flex-${direction} flex-${wrapping} justify-${justify} items-${align} p-${padding} m-${margin}`}
  >
    {children}
  </div>
)

Flex.defaultProps = {
  display: 'flex',
  direction: 'col',
  wrapping: 'wrap',
  justify: 'start',
  align: 'start',
  padding: '0',
  margin: '0'
}

Flex.propTypes = {
  display: PropTypes.oneOf(['flex', 'inline-flex']),
  direction: PropTypes.oneOf(['row', 'row-reverse', 'col', 'col-reverse']),
  wrapping: PropTypes.oneOf(['no-wrap', 'wrap', 'wrap-reverse']),
  justify: PropTypes.oneOf(['start', 'center', 'end', 'between', 'around']),
  align: PropTypes.oneOf(['stretch', 'start', 'center', 'end', 'baseline']),
  padding: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  margin: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  children: PropTypes.node.isRequired,
  smHeight: PropTypes.string,
  mdHeight: PropTypes.string,
  lgHeight: PropTypes.string,
  xlHeight: PropTypes.string,
  smWidth: PropTypes.string,
  mdWidth: PropTypes.string,
  lgWidth: PropTypes.string,
  xlWidth: PropTypes.string
}

export default Flex
