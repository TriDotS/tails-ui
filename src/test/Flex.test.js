import Flex from '../components/Flex'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Flex>A flex box</Flex>, div)
})

test('flex snapshot', () => {
  const tree = renderer.create(<Flex>A flex box</Flex>).toJSON()
  expect(tree).toMatchSnapshot()
})
