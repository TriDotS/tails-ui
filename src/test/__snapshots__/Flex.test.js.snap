// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flex snapshot 1`] = `
<div
  className="flex flex-col flex-wrap justify-start items-start p-0 m-0"
>
  A flex box
</div>
`;
